examples/viewer_lib/ui/segmentation/segment_editor_ui.py

```diff
@@ -22,142 +25,185 @@
-                self._register_effect_ui(SegmentationEffectErase, PaintEffectUI)
-                self._register_effect_ui(SegmentationEffectIslands, IslandsEffectUI)
-
-            with VRow():
-                VBtn(
-                    variant="text",
-                    density="compact",
-                    text="Opacity",
-                    prepend_icon=(
-                        f"{self._typed_state.name.segment_opacity.is_visible} ? 'mdi-menu-down' : 'mdi-menu-right' ",
-                    ),
-                    click=f"{self._typed_state.name.segment_opacity.is_visible} = !{self._typed_state.name.segment_opacity.is_visible}",
-                    style="text-transform: none;",
+                SegmentationRenderingUI(
+                    typed_state=self.sub_state(self._typed_state.name.segment_rendering),
+                    variant="flat",
                 )
 
-                self._opacity_ui = SegmentationOpacityUI(
-                    typed_state=self.sub_state(self._typed_state.name.segment_opacity),
-                    v_if=self._typed_state.name.segment_opacity.is_visible,
+    def build_toolbar_ui(self, **kwargs):
+        with FlexContainer(classes="flex-grow-1", **kwargs):
+            VDivider(classes="my-2")
+            VSpacer()
+            self._build_effect_buttons(
+                v_if=(f"!{self._viewer_state.name.is_drawer_visible}",),
+                all=False,
+            )
+            VSpacer()
+            VDivider(classes="my-2")
+            self._build_undo_redo_buttons()
+
+    def _build_effect_buttons(self, all: bool = True, **kwargs):
+        with FlexContainer(**kwargs):
+            self._create_effect_button(
+                "No tool",
+                "mdi-cursor-default",
+                SegmentationEffectNoTool,
+            )
+            self._create_effect_button(
+                "Paint",
+                "mdi-brush",
+                SegmentationEffectPaint,
+            )
+            self._create_effect_button(
+                "Erase",
+                "mdi-eraser",
+                SegmentationEffectErase,
+            )
+            self._create_effect_button(
+                "Scissors",
+                "mdi-content-cut",
+                SegmentationEffectScissors,
+            )
+            if all:
+                self._create_effect_button(
+                    "Threshold",
+                    "mdi-auto-fix",
+                    SegmentationEffectThreshold,
+                )
+                self._create_effect_button(
+                    "Islands",
+                    "mdi-scatter-plot",
+                    SegmentationEffectIslands,
                 )
-                VDivider()
 
-    def _register_effect_ui(self, effect_type: type[SegmentationEffect], effect_ui_type: type):
-        self._effect_ui[effect_type] = effect_ui_type(v_if=self.button_active(effect_type))
+    def _build_undo_redo_buttons(self, **kwargs):
```

Comment:
Undo / redo buttons shouldn't be in the segmentation UI but directly in the application UI level toolbar